spidermon/contrib/scrapy/pipelines.py

```diff
@@ -168,3 +181,19 @@ def _add_error_stats(self, errors):
             for message in messages:
                 self.stats.add_field_error(field_name, message)
         self.stats.add_item_with_errors()
+
+    def _close_spider_by_stats(self, spider, stats_dict):
+        """
+        This method evaluate each stat value and if no of errors are greater then
+        max_errors_allowed. Spider will be closed gracefully.
+        """
+        for stat_name, max_errors_allowed in stats_dict.iteritems():
+            stat_current_value = self.stats.stats.get_value(stat_name, 0)
+
+            if stat_current_value > max_errors_allowed:
+                self.crawler.engine.close_spider(
+                    spider,
+                    'Spidermon: No of errors are greater then the expected no of errors {} > {}'.format(
```

Comment:
I think we should not call them errors (they could be warnings or something else entirely), and I would include the name of the specific stat that triggers the spider closing.

```suggestion
                    'Spidermon: The value of stat {} ({}) went above the configured limit ({})'.format(stat_name, stat_current_value, max_errors_allowed)
```